src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -197,6 +212,18 @@ var DOMRenderer = ReactFiberReconciler({
     return true;
   },
 
+  commitMount(
+    domElement : Instance,
+    type : string,
+    newProps : Props,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : void {
+    if (shouldAutoFocusHostComponent(type, newProps)) {
```

Comment:
Hm, I think you're right, but there's this: https://github.com/facebook/react/blob/3bc5595dfd6db741b19f17e7aef98b2c1e4be847/src/renderers/shared/fiber/ReactFiberCompleteWork.js#L240-L243

I think this is actually not intentional. I'll follow up on it. 😄